demo/app/src/components/token-send/TokenSend.tsx

```diff
@@ -0,0 +1,340 @@
+      vaultHexChainCode,
+      walletCore.CoinTypeExt.derivationPath(walletCore.CoinType.thorchain),
+      false
+    );
+    const pubkeyData = Buffer.from(thorPublicKey, "hex");
+    const publicKey = walletCore.PublicKey.createWithData(
+      pubkeyData,
+      walletCore.PublicKeyType.secp256k1
+    );
+
+    try {
+      const hashes = walletCore.TransactionCompiler.preImageHashes(
+        walletCore.CoinType.thorchain,
+        inputData
+      );
+      const preSigningOutput =
+        TW.TxCompiler.Proto.PreSigningOutput.decode(hashes);
+      const allSignatures = new walletCore.DataVector();
+      const allPublicKeys = new walletCore.DataVector();
+      const signature = getSignatureWithRecoveryID(
+        signatures[byteArrayToHexString(preSigningOutput.dataHash)]
+      );
+      console.log(
+        222,
+        preSigningOutput,
+        preSigningOutput.dataHash,
+        byteArrayToHexString(preSigningOutput.dataHash)
+      );
+      console.log(333, "signature", signature);
+      if (!publicKey.verify(signature, preSigningOutput.dataHash)) {
+        console.error("fail to verify signature");
+        return;
+      }
+
+      allSignatures.add(signature);
+      allPublicKeys.add(publicKey.data());
+
+      const compileWithSignature =
+        walletCore.TransactionCompiler.compileWithSignatures(
+          walletCore.CoinType.thorchain,
+          inputData,
+          allSignatures,
+          allPublicKeys
+        );
+      const output = TW.Cosmos.Proto.SigningOutput.decode(compileWithSignature);
+
+      {
+        const buffer = Buffer.from(output.serialized, "base64");
+        const hash = createHash("sha256").update(buffer).digest("hex");
+        console.log("txHash:", hash);
+      }
+
+      return output.serialized; // raw transaction
+    } catch (error: any) {
+      console.error(
+        `fail to get signed ethereum transaction, error: ${error.message}`
+      );
+      return;
+    }
+  };
+
+  const sendTransaction = async (
+    vaultPublicKeyEcdsa: string,
+    vaultLocalPartyId: string,
+    vaultHexChainCode: string,
+    fromAddress: string,
+    toAddress: string,
+    amount: string,
+    passwd: string
+  ) => {
+    if (
+      !vaultPublicKeyEcdsa ||
+      !vaultLocalPartyId ||
+      !vaultHexChainCode ||
+      !fromAddress ||
+      !toAddress ||
+      !amount
+    )
+      return;
+
+    try {
+      const walletCore = await initWasm();
+      const accountInfo = await (await getAccountInfo(fromAddress)).json();
+      const accountNumber = accountInfo.result.value.account_number;
+      const sequence = accountInfo.result.value.sequence;
+      const thorchainspecific = new THORChainSpecific({
+        account_number: accountNumber,
+        sequence: sequence,
+        fee: 2000000,
+      });
+      const coin = new Coin({
+        chain: "THORChain",
+        ticker: "RUNE",
+        decimals: 8,
+        is_native_token: true,
+        hex_public_key: vaultPublicKeyEcdsa,
```

Comment:
you can't put `vaultPublicKeyEcdsa` here , because it is the vault's public key ecdsa , but not for the coin, 

instead you should put the public key you get from https://github.com/vultisig/vultisigner/pull/60/files#diff-e44767e3d17c37ae4bb61ce0af3f2b76a13752bd9c17d5481a3fe1e279db08cbR98